packages/staking/src/features/store/useDelegationPortfolioStore.ts

```diff
@@ -0,0 +1,231 @@
+import { DelegatedStake } from '@cardano-sdk/wallet';
+import { Wallet } from '@lace/cardano';
+import { create } from 'zustand';
+import { immer } from 'zustand/middleware/immer';
+import { MAX_POOLS_COUNT } from './delegationPortfolio';
+import { CurrentPortfolioStakePool, DraftPortfolioStakePool } from './types';
+
+enum Mode {
+  Overview = 'Overview',
+  BrowsePools = 'BrowsePools',
+  CurrentPoolDetails = 'CurrentPoolDetails',
+  PoolDetails = 'PoolDetails',
+  CurrentPortfolioManagement = 'CurrentPortfolioManagement',
+  NewPortfolioCreation = 'NewPortfolioCreation',
+}
+
+enum DrawerDefaultSection {
+  DETAIL = 'detail',
+}
+
+enum DrawerManagementSection {
+  PREFERENCES = 'preferences',
+  CONFIRMATION = 'confirmation',
+  SIGN = 'sign',
+  SUCCESS_TX = 'success_tx',
+  FAIL_TX = 'fail_tx',
+}
+
+type Section = DrawerDefaultSection | DrawerManagementSection;
+
+enum OverviewCommand {
+  ShowDetails = 'ShowDetails',
+  Manage = 'Manage',
+  GoToBrowsePools = 'GoToBrowsePools',
+}
+
+enum BrowsePoolsCommand {
+  Select = 'Select',
+  ShowDetails = 'ShowDetails',
+}
+
+enum PoolDetailsCommand {
+  Select = 'Select',
+}
+
+enum DrawerPreferencesCommand {
+  UpdateWeight = 'UpdateWeight',
+}
```

Comment:
I'd consider coupling the commands with the expected data and making it a union of objects, otherwise it gets pretty tricky for the reader to understand what are the commands parametrized by which I believe is important to get the bigger picture

```suggestion
type DrawerPreferencesCommand = {
  name: 'UpdateWeight',
  params: { weight: number }
} | ...
```

I believe this should also make it pretty straightforward to come up with the "command" callback signature below

```
// WIP: trying to match command to its corresponding data/payload

```